src/collection/set_collection.py

```diff
@@ -68,8 +68,20 @@ def upsert_to_collection_from_vectors(
         collection_name (str): name of collection
         data (list[PointStruct]): vectors to upsert
     """
-    print(f"Upserting {len(data)} points to collection {collection_name}...")
-    operation_info = client.upsert(
-        collection_name=collection_name, wait=True, points=data
-    )
-    print(operation_info)
+
+    def _chunk_data(data, chunk_size=100):
+        """Chunk data into smaller pieces for upsertion"""
+        for i in range(0, len(data), chunk_size):
+            yield data[i : i + chunk_size]
+
+    chunk_size = 500
```

Comment:
One for the future - we might consider moving this to a config file, along with other params (similarity score threshold?)

Reply:
Issue #17